posthog/queries/session_recordings/test/__snapshots__/test_session_recording_list.ambr

```diff
@@ -1172,3 +1052,1109 @@
   OFFSET 0
   '
 ---
+# name: TestClickhouseSessionRecordingsListV2.test_event_filter
+  '
+  
+  SELECT session_recordings.session_id,
+         start_time,
+         end_time,
+         click_count,
+         keypress_count,
+         urls,
+         duration,
+         session_recordings.distinct_id as distinct_id ,
+         matches_0 ,
+         matching_events_0
+  FROM
+    (SELECT session_id,
+            distinct_id ,
+            groupUniqArrayIf(100)((timestamp,
+                                   uuid,
+                                   session_id,
+                                   window_id), event_match_0 = 1) AS matching_events_0 ,
+            sum(event_match_0) AS matches_0
+     FROM
+       (SELECT uuid,
+               distinct_id,
+               event,
+               team_id,
+               timestamp,
+               $session_id as session_id,
+               $window_id as window_id ,
+               if(event = '$pageview', 1, 0) as event_match_0
+        FROM events
+        WHERE team_id = 2
+          AND event IN ['$pageview']
+          AND timestamp >= '2021-01-13 12:00:00'
+          AND timestamp <= '2021-01-22 08:00:00'
+          AND notEmpty(session_id) )
+     GROUP BY session_id,
+              distinct_id
+     HAVING 1 = 1
+     AND matches_0 > 0) AS events
+  JOIN
+    (SELECT session_id,
+            any(window_id) as window_id,
+            minIf(first_event_timestamp, first_event_timestamp != '1970-01-01 00:00:00') as start_time,
+            MAX(last_event_timestamp) as end_time,
+            SUM(click_count) as click_count,
+            SUM(keypress_count) as keypress_count,
+            groupArrayArray(urls) as urls,
+            dateDiff('second', start_time, end_time) as duration,
+            any(distinct_id) as distinct_id,
+            SUM(has_full_snapshot) as full_snapshots
+     FROM session_recording_events PREWHERE team_id = 2
+     AND timestamp >= '2021-01-13 12:00:00'
+     AND timestamp <= '2021-01-22 08:00:00'
+     GROUP BY session_id
+     HAVING full_snapshots > 0
+     AND start_time >= '2021-01-14 00:00:00'
+     AND start_time <= '2021-01-21 20:00:00') AS session_recordings ON session_recordings.session_id = events.session_id
+  WHERE session_recordings.distinct_id == events.distinct_id
```

Comment:
doesn't matching by distinctID lead to problems as we're randomly selecting one?

Don't think sessionIDs change between login steps, but distinctIDs can & do?

Reply:
I also wonder this... Seems like a copy from the v1 query but I'm not sure why it matters... We only need one distinctId to load the Person anyways and we have that from the recording. We anyways group by session_id... 

Perhaps this is to account for weird cases where there is a clash of session_id but I don't see that happening...